From 715bada766d6d410b7bdbaf7fc2086f7a61c8199 Mon Sep 17 00:00:00 2001
From: Maarten Brock <m.brock@vanmierlo.com>
Date: Wed, 4 Dec 2019 09:58:55 +0530
Subject: [PATCH 0030/1851] serial: xilinx_uartps: set_mctrl sets RTS and DTR

commit 6c68297c7365e70a0b11632e03c8895d238f0789 from
https://github.com/Xilinx/linux-xlnx.git

set_mctrl now sets RTS and DTR.

Signed-off-by: Maarten Brock <m.brock@vanmierlo.com>
Signed-off-by: Shubhrajyoti Datta <shubhrajyoti.datta@xilinx.com>
Link: https://lkml.org/lkml/2019/11/25/318
State: upstream (1d3c2ea44b1163269062c891d073f4764686028a)
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/tty/serial/xilinx_uartps.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/tty/serial/xilinx_uartps.c b/drivers/tty/serial/xilinx_uartps.c
index 45ccb5528dce..12aaac1b7d39 100644
--- a/drivers/tty/serial/xilinx_uartps.c
+++ b/drivers/tty/serial/xilinx_uartps.c
@@ -1058,6 +1058,10 @@ static void cdns_uart_set_mctrl(struct uart_port *port, unsigned int mctrl)
 	val &= ~(CDNS_UART_MODEMCR_RTS | CDNS_UART_MODEMCR_DTR);
 	mode_reg &= ~CDNS_UART_MR_CHMODE_MASK;
 
+	if (mctrl & TIOCM_RTS)
+		val |= CDNS_UART_MODEMCR_RTS;
+	if (mctrl & TIOCM_DTR)
+		val |= CDNS_UART_MODEMCR_DTR;
 	if (mctrl & TIOCM_LOOP)
 		mode_reg |= CDNS_UART_MR_CHMODE_L_LOOP;
 	else
-- 
2.31.1

